src/Gaffer/TweakPlug.cpp

```diff
@@ -247,7 +247,19 @@ bool TweakPlug::applyTweak( IECore::CompoundData *parameters, MissingMode missin
 			{
 				return parameters->writable().erase( valueName ) > 0;
 			}
-			parameters->writable()[valueName] = newData;
+
+			DataPtr &dst = parameters->writable()[valueName];
+
+			if( runTimeCast<InternedStringData>( dst.get() ) )
+			{
+				if( const StringData *s = runTimeCast<const StringData>( newData.get() ) )
+				{
+					dst = new InternedStringData( s->readable() );
+					return true;
+				}
+			}
```

Comment:
Sorry, I should have spotted this in an earlier review. Doing the StringData->InternedStringData conversion here means that it only works when tweaking `CompoundData`, not for any tweaks applied using custom functors (like your OptionTweaks for instance). Would it be better to revert the change here and do the StringData->InternedStringData conversion in TweakPlug.inl, where you've currently just modified the check? That also has the benefit of keeping the type logic all in one place, instead of it needing to match in two different parts of the code.

Reply:
No problem at all - moving that made it simpler overall and I could remove the awkward `!( currentValue->typeId() == IECore::StringData::staticTypeId() && newData->typeId() == IECore::InternedStringData::staticTypeId() )` test.

I think it's ready to look at in 6fb231c9daa7f12413e154dd49e2b452e460278d